services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy

```diff
@@ -879,6 +879,70 @@ class FlowCrudSpec extends HealthCheckSpecification {
+        given: "Two active not neighboring switches"
+        def switchPair = topologyHelper.getAllNotNeighboringSwitchPairs().find {
+            it.paths.find { it.unique { it.switchId }.size() >= 4 }
+        } ?: assumeTrue("No suiting switches found", false)
+
+        and: "Select path for further manipulation with it"
+        def selectedPath = switchPair.paths.max { it.size() }
+        def altPaths = switchPair.paths.findAll { it != selectedPath }
+
+        and: "Make all alternative paths unavailable (bring ports down on the src/intermediate switches)"
+        List<PathNode> broughtDownPortsSrcSwitch = []
+        altPaths.findAll { it.first().portNo != selectedPath.first().portNo }.unique { it.first() }.each { path ->
+            def src = path.first()
+            broughtDownPortsSrcSwitch.add(src)
+            antiflap.portDown(src.switchId, src.portNo)
+        }
+
+        List<PathNode> broughtDownPortsIntermSwitch = []
+        altPaths.findAll { it.first().portNo == selectedPath.first().portNo &&
+                it[2].portNo != selectedPath[2].portNo && it[2].switchId == selectedPath[2].switchId
+        }.unique { it[2] }.each { path ->
+            def src = path[2]
+            broughtDownPortsIntermSwitch.add(src)
+            antiflap.portDown(src.switchId, src.portNo)
+        }
+
+        Wrappers.wait(antiflapMin + WAIT_OFFSET) {
+            assert northbound.getAllLinks().findAll {
+                it.state == FAILED
+            }.size() == (broughtDownPortsSrcSwitch.size() + broughtDownPortsIntermSwitch.size()) * 2
+        }
+
+        and: "Make reverse path not preferable. Forward path is still preferable(on the intermediate switch)"
+        def flowBandwidth = 500  // flow bandwidth should be bigger than newIslBandwidth
+        def islsToModify = pathHelper.getInvolvedIsls(selectedPath)[1]
+        def newIslBandwidth = 100
+        islsToModify.each {
+            database.updateIslAvailableBandwidth(it.reversed, newIslBandwidth)
+            database.updateIslMaxBandwidth(it.reversed, newIslBandwidth)
+        }
+
+        when: "Create a flow"
+        def flow = flowHelper.randomFlow(switchPair)
+        flow.maximumBandwidth = flowBandwidth
+        flowHelper.addFlow(flow)
+
+        then: "Flow is not created"
+        def e = thrown(HttpClientErrorException)
+        e.statusCode == HttpStatus.NOT_FOUND
+        e.responseBodyAsString.to(MessageError).errorMessage.contains("Could not create flow")
+
+        and: "Restore topology, delete the flow and reset costs"
+        [broughtDownPortsSrcSwitch, broughtDownPortsIntermSwitch].each { sw ->
+            sw.each { antiflap.portUp(it.switchId, it.portNo) }
+        }
+        Wrappers.wait(discoveryInterval + WAIT_OFFSET) {
+            northbound.getAllLinks().each { assert it.state != FAILED }
+        }
+        database.resetCosts()
+        islsToModify.each { database.resetIslBandwidth(it) }
+    }
```

Comment:
I think we should prefer adding new tests to V2 specification instead of V1

Reply:
agree.

this test was requested by dev team especially for v1